app/preact/src/server/framework-preset-preact.ts

```diff
@@ -9,3 +11,20 @@ export function babelDefault(config: TransformOptions) {
     ],
   };
 }
+
+export function webpackFinal(config: Configuration) {
+  return {
+    ...config,
+    resolve: {
+      ...config.resolve,
+      // Prefer top-level aliases
+      modules: [path.resolve('node_modules'), ...config.resolve.modules],
+      // Alias react imports to "preact/compat":
+      alias: {
+        ...config.resolve.alias,
+        react: 'preact/compat',
+        'react-dom': 'preact/compat',
```

Comment:
```suggestion
        react: require.resolve('preact/compat'),
        'react-dom': require.resolve('preact/compat'),
```
These should be resolved